statsmodels/base/transform.py

```diff
@@ -0,0 +1,218 @@
+        transformation using method.
+
+        Parameters
+        ----------
+        x : array_like
+            The untransformed data.
+        bounds : tuple
+            Numeric 2-tuple, that indicate the solution space for the lambda
+            parameter. Default (-1, 2).
+        R : int
+            The seasonality/grouping parameter. Default 2.
+        method : {'guerrero', 'loglik'}
+            The method by which to estimate lambda. Defaults to 'guerrero', but
+            the profile likelihood ('loglik') is also available.
+        scale : {'sd', 'mad'}
+            The dispersion measure to be used. 'sd' indicates the sample standard
+            deviation, but the more robust 'mad' is also available.
+
+        Returns
+        -------
+        lmbda : float
+            The lambda parameter.
+        """
+        method = method.lower()
+
+        if len(bounds) != 2:
+            raise ValueError("Bounds of length {0} not understood."
+                             .format(len(bounds)))
+        elif bounds[0] >= bounds[1]:
+            raise ValueError("Lower bound exceeds upper bound.")
+
+        if method == 'guerrero':
+            lmbda = self._guerrero_cv(x, R, bounds, scale=scale)
+        elif method == 'loglik':
+            lmbda = self._loglik(x, bounds)
+        else:
+            raise ValueError("Method '{0}' not understood.".format(method))
+
+        return lmbda
+
+    def _guerrero_cv(self, x, R, bounds, scale='sd'):
+        """
+        Computes guerrero's coefficient of variation. If no seasonality
+        is present in the data, R is set to 2 (p. 40, comment).
+
+        NOTE: Seasonality-specific auxiliaries *should* provide their own
+        seasonality parameter.
+
+        Parameters
+        ----------
+        x : array_like
+        R : int
+            Seasonality/grouping parameter. Default 2.
+        bounds : tuple
+            Numeric 2-tuple, that indicate the solution space for the lambda
+            parameter.
+        scale : {'sd', 'mad'}
+            The dispersion measure to be used. 'sd' indicates the sample standard
+            deviation, but the more robust 'mad' is also available.
+        """
+        nobs = len(x)
+        groups = int(nobs / R)
+        scale = scale.lower()
+
+        # remove the first n < R observations from consideration.
+        grouped_data = np.reshape(x[nobs - (groups * R): nobs], (R, groups))
+
+        mean = np.mean(grouped_data, 1)
+
+        if scale == 'sd':
+            dispersion = np.std(grouped_data, 1)
+        elif scale == 'mad':
+            dispersion = mad(grouped_data, axis=1)
+        else:  # TODO: IQR?
+            raise ValueError("Scale '{0}' not understood.".format(scale))
+
+        # closure; it's more elegant - and efficient - this way
+        def optim(lmbda, *args, **kwargs):
+            rat = np.divide(dispersion, np.power(mean, 1 - lmbda))  # eq. 6, p. 40
+            return np.std(rat) / np.mean(rat)
```

Comment:
Yes, I think CV would be more appropriate than MAD based in most cases. The grouped MAD should make it insensitive to local outliers, and we want CV to capture the large variation, and not ignore parts of it like MAD would do.